sktree/ensemble/_honest_forest.py

```diff
@@ -451,14 +453,23 @@ def _predict_proba(self, X, indices=None, impute_missing=None):
         )
 
         # Normalize to unit length, due to prior weighting
-        zero_mask = posteriors.sum(1) == 0
+        posteriors = np.array(posteriors)
+        zero_mask = posteriors.sum(2) == 0
         posteriors[~zero_mask] /= posteriors[~zero_mask].sum(1, keepdims=True)
+
         if impute_missing is None:
             posteriors[zero_mask] = self.empirical_prior_
```

Comment:
Having trouble on this line. It seems that when defined, `empirical_prior_` doesn't have dimensional differences for `y` labels. So I'll try to increase its dimensions.

Reply:
Actually, why is the empirical prior set at the forest level. It's already set in honest trees, so this part can be removed.